plugin/entryBase.go

```diff
@@ -0,0 +1,87 @@
+package plugin
+
+import (
+	"flag"
+	"time"
+)
+
+type cacheableOp int8
+
+const (
+	// List represents Group#List
+	List cacheableOp = iota
+	// Open represents Readable#Open
+	Open
+	// Metadata represents Resource#Metadata
+	Metadata
+)
+
+var cacheableOpToNameMap = [3]string{"List", "Open", "Metadata"}
+
+// EntryBase implements Entry, making it easy to create new entries.
+// You should use plugin.NewEntry to create new EntryBase objects.
+type EntryBase struct {
+	name string
+	// id represents the entry's wash ID. It is set in CachedList.
+	id  string
+	ttl [3]time.Duration
+}
+
+// NewEntry creates a new entry
+func NewEntry(name string) EntryBase {
```

Comment:
I could move `NewEntry` back to `types.go` and put the tests in a `types_test.go` file if that's preferred.

Reply:
This seems ok. The documentation will all be rolled up, and it's enough extra stuff that it would obfuscate the other things in types.go.